release-process.rst

```diff
@@ -28,51 +34,91 @@ Roughly:
+   -  Extensions support MAY be ended (moved to PECL)
+   -  Backward compatibility MAY be broken
+   -  API compatibility MAY be broken (internals and userland)
+   -  ABI MAY be broken (internals)
 
 Minor Version Number
 ====================
 
 -  x.y.z to x.y+1.z
 
-   -  Bugfixes
-   -  New features
-   -  Extensions support can be ended (moved to PECL)
-   -  Backward compatibility must be kept
-   -  API compatibility must be kept (userland)
-   -  ABI and API can be broken (internals)
-   -  Source compatibility should be kept if possible, while breakages are
+   -  It SHOULD included bugfixes and new features.
+
+   -  Extensions support MAY be ended (moved to PECL).
+
+   -  Syntax backward compatibility SHOULD be preserved - every PHP program that
+      compiles SHOULD continue to compile.
+
+   -  Backwards compatibility breaks in minor versions MUST NOT result in silent
+      behavioral differences. Instead any breaking change MUST be "obvious" when
+      executing the program.
+
+   -  API backward compatibility breaks SHOULD be limited to extensions, or the
+      API of functions within an extension
+
+   -  ABI and internal API compatibility breaks are NOT RECOMMENDED.
+
+   -  Source compatibility SHOULD be kept if possible, while breakages are
       allowed
 
 Patch Version Number
 ====================
 
 -  x.y.z to x.y.z+1
 
-   -  Bug fixes and security patches only
-   -  Extensions support can't be removed (like move them to PECL)
-   -  Backward compatibility must be kept (internals and userland)
-   -  ABI and API compatibility must be kept (internals)
+   -  It SHOULD included bug fixes and security patches
+   -  New features MUST NOT be added.
+   -  Extensions support MUST NOT be removed (like move them to PECL)
+   -  Backward compatibility MUST be kept (internals and userland)
+   -  ABI and internal API compatibility SHOULD be preserved for high severity
+      security issues, and MUST be preserved for all other security issues.
 
 It is critical to understand the consequences of breaking BC, APIs or ABIs (only
-internals related). It should not be done for the sake of doing it. RFCs
+internals related). It SHOULD NOT be done for the sake of doing it. RFCs
 explaining the reasoning behind a breakage and the consequences along with test
 cases and patch(es) should help.
 
+If a high severity security fix requires breaking the internal ABI or API, a
+proper migration path MUST be provided, and the impact MUST be minimized as much
+as possible. This MUST also be accompanied by additional communication during
+the release.
+
+All new user-facing features MUST be mentioned in the `UPGRADING
+<https://github.com/php/php-src/blob/master/UPGRADING>`_ document. All API and
+ABI breaks MUST be mentioned in the `UPGRADING.INTERNALS
+<https://github.com/php/php-src/blob/master/UPGRADING.INTERNALS>`_ document.
+
 See the following links for explanation about API and ABI:
```

Comment:
There are few terms of compatibility in this document:
- ABI
- API internal
- API userland
- syntax.

Can all of these be briefly explained (for new contributors) in section Backward Compatibility and part starting on line 122 (old 76) may be removed?